testing/fetcher.ts

```diff
@@ -329,9 +329,9 @@ export class AuthenticatingFetcher implements Fetcher {
             );
           }
 
-          if (bodyWithSubstitutions) {
+          if (typeof bodyWithSubstitutions === 'string') {
             bodyWithSubstitutions = replaceAll(
-              bodyWithSubstitutions,
+              bodyWithSubstitutions as string,
```

Comment:
Are you sure you need the `as string` here? TS should be able to infer it. Same above.

Reply:
for some reason TS was not able to infer the type. I thought it was VS code but `git push` also complains on the linter part.

Reply:
![image](https://user-images.githubusercontent.com/70549840/151088903-5fd7912a-7d36-42db-9d09-c1e8c3df4df6.png)

Reply:
```
huayang@Huayangs-MBP [17:40:15] [~/code/packs-sdk] [hg-buffer-body *]
-> % make build 
testing/fetcher.ts:296:89 - error TS2339: Property 'replace' does not exist on type 'string | Buffer'.
  Property 'replace' does not exist on type 'Buffer'.

296             bodyWithTemplateSubstitutions = ensureExists(bodyWithTemplateSubstitutions).replace(
                                                                                            ~~~~~~~


Found 1 error.

make: *** [compile] Error 2
```

Reply:
actually, the error is for the one above. this one can be removed.